src/go/MONIT/annotationManager.go

```diff
@@ -0,0 +1,595 @@
+	array = append(array, month)
+	array = append(array, day)
+	array = append(array, hour)
+	array = append(array, min)
+	array = append(array, sec)
+
+	return
+}
+
+//
+// The following block of code was taken from
+// https://github.com/dmwm/CMSMonitoring/blob/master/src/go/MONIT/alert.go#L259
+//Helper function for time difference between two time.Time objects
+func timeDiffHelper(timeList []int) (dif string) {
+	for ind := range timeList {
+		if timeList[ind] > 0 {
+			switch ind {
+			case 0:
+				dif += strconv.Itoa(timeList[ind]) + "Y "
+				break
+			case 1:
+				dif += strconv.Itoa(timeList[ind]) + "M "
+				break
+			case 2:
+				dif += strconv.Itoa(timeList[ind]) + "D "
+				break
+			case 3:
+				dif += strconv.Itoa(timeList[ind]) + "h "
+				break
+			case 4:
+				dif += strconv.Itoa(timeList[ind]) + "m "
+				break
+			case 5:
+				dif += strconv.Itoa(timeList[ind]) + "s "
+				break
+			default:
+				break
+			}
+		}
+	}
+
+	return
+}
+
+//
+// The following block of code was taken from
+// https://github.com/dmwm/CMSMonitoring/blob/master/src/go/MONIT/alert.go#L291
+//Function for time difference between two time.Time objects
+func timeDiff(t1 time.Time, t2 time.Time, duration int) string {
+	if t1.After(t2) {
+		timeList := diff(t1, t2)
+		return timeDiffHelper(timeList) + "AGO"
+	}
+
+	timeList := diff(t2, t1)
+	if duration == 1 {
+		return timeDiffHelper(timeList)
+	}
+	return "IN " + timeDiffHelper(timeList)
+
+}
+
+//
+// The following block of code was taken from (with few changes done)
+// https://github.com/dmwm/CMSMonitoring/blob/master/src/go/MONIT/alert.go#L426
+//Function for printing annotations in Plain text format
+func tabulate(data []annotationData) {
+
+	w := new(tabwriter.Writer)
+	w.Init(os.Stdout, 8, 8, 0, '\t', 0)
+	defer w.Flush()
+
+	fmt.Fprintf(w, "\n ")
+	for _, each := range configJSON.Columns {
+		fmt.Fprintf(w, "%s\t", each)
+	}
+	fmt.Fprintf(w, "\n")
+
+	for _, each := range data {
+
+		fmt.Fprintf(w, " %d\t%d\t%s\t%s\t%s",
+			each.ID,
+			each.DashboardID,
+			strings.Split(each.Text, "\n<a")[0],
+			each.Tags,
+			timeDiff(time.Now(), time.Unix(0, each.Time*int64(microSec)), 0), // times should be in microseconds of Unix since epoch
+		)
+		if each.TimeEnd == maxtstmp {
+			fmt.Fprintf(w, "\t%s", "Undefined")
+			fmt.Fprintf(w, "\t%s\n", "Undefined")
+		} else {
+			fmt.Fprintf(w, "\t%s", timeDiff(time.Now(), time.Unix(0, each.TimeEnd*int64(microSec)), 0))                                // times should be in microseconds of Unix since epoch
+			fmt.Fprintf(w, "\t%s\n", timeDiff(time.Unix(0, each.Time*int64(microSec)), time.Unix(0, each.TimeEnd*int64(microSec)), 1)) // times should be in microseconds of Unix since epoch
+		}
+	}
+}
+
+//Function running all logics
+func run() {
+
+	var aData []annotationData
+	tagList := parseTags(tags)
+	err := getAnnotations(&aData, tagList)
+	if err != nil {
+		log.Fatalf("Unable to fetch annotations, error :%v", err)
+	}
+	tabulate(aData)
+
+	switch action {
```

Comment:
the action switch should provide all cases: list, create, update, delete. So far I only see list (which is getAnnotations) and delete. Can you implement create and update?

Reply:
Valentin as far as I see, monit tool already has the feature to create an annotation, and also our intelligence module is creating automatically. Do we also need to add it to this tool? 
Yes, update can be implemented.

Reply:
So, it seems to me that we should review both annotation and alert tools to decide. So far I want to complete this one to cover all actions we need, then we can decide which functionality should be kept where. The alert tool should probably only handle alerts, while annotations tool should handle only annotations.

Reply:
Done !